src/libraries/System.IO.Compression/tests/ZipArchive/zip_LargeFiles.cs

```diff
@@ -126,4 +133,85 @@ public static void CheckZIP64VersionIsSet_ForSmallFilesAfterBigFiles(bool isComp
             File.Delete(zipArchivePath);
         }
     }
+
+    [ConditionalFact(typeof(PlatformDetection), nameof(PlatformDetection.IsSpeedOptimized), nameof(PlatformDetection.Is64BitProcess))] // don't run it on slower runtimes
+    [OuterLoop("It requires around 6 GB of free disk space")]
+    public static void CompatZip64BeforeAndAfterFix()
+    {
+        // This test has the purpose of confirming that ZipArchive can still process a zip file that was created
+        // with these APIs before the Zip64 bug was fixed: https://github.com/dotnet/runtime/pull/102053
+
+        ushort buggyZip64Version = 20;
+        byte[] largeBuffer = GC.AllocateUninitializedArray<byte>(1_000_000_000); // 1 GB
+        string zipArchivePath = Path.Combine(Path.GetTempPath(), "over4GB.zip");
+
+        try
+        {
+            using FileStream fs = File.Open(zipArchivePath, FileMode.Create, FileAccess.ReadWrite);
+
+           // Create
+            using (ZipArchive archive = new(fs, ZipArchiveMode.Create, true))
+            {
+                ZipArchiveEntry file = archive.CreateEntry("file.txt", CompressionLevel.NoCompression);
+
+                using (Stream stream = file.Open())
+                {
+                    // Write 5GB of data
+                    for (var i = 0; i < 5; i++)
+                    {
+                        stream.Write(largeBuffer);
+                    }
+                }
+            }
+            Assert.True(fs.Length > int.MaxValue, $"File size is not big enough to test the Zip64 fix: {fs.Length} vs {int.MaxValue}");
+
+            fs.Position = 0;
+
+            // Open archive to modify the bit as it used to look before fix
+            using (ZipArchive zip = new ZipArchive(fs, ZipArchiveMode.Read, leaveOpen: true))
+            {
+                FieldInfo offsetOfLHField = GetOffsetOfLHField();
+
+                ZipArchiveEntry entry = zip.Entries.First();
+
+                long currentPosition = fs.Position;
+
+                // Confirm it's initially set to the correct value
+                using (BinaryReader reader = new(fs, Encoding.UTF8, leaveOpen: true))
```

Comment:
You can open the reader and writer at the same time, and just move the position back 2 bytes after you read it.  That makes this more reabable and should reduce the code a bit.

Reply:
I think it is more important to be able to close the ZipArchive, dispose it, then reopen it from the same stream, and confirm that the ZipArchive can be opened, the ZipArchiveEntry can be created without unexpected errors, and finally confirm the zip64 value was unmodified.

If I reduce the code as suggested, that would mean using the exact same ZipArchive object, without really verifying that the ZipArchive and the ZipArchiveEntry instances could still be created successfully.